src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,735 @@
+            pf.write(f"#{'=' * 50}\n")
+            pf.write(f"# config_file = {self.telescope_model.config_file_path}\n")
+            pf.write(f"# zenith_angle [deg] = {self.ZENITH_ANGLE_DEG}\n")
+            pf.write(
+                f"# off_axis_angle [deg] = {self.config_data['off_axis_angle'].to_value(u.deg)}\n"
+            )
+            pf.write(f"# source_distance [km] = {self.config_data['source_distance']}\n")
+
+        with stars_file.open("w", encoding="utf-8") as sf:
+            zen = self.ZENITH_ANGLE_DEG
+            dist = float(self.config_data["source_distance"])
+            sf.write(f"0. {90.0 - zen} 1.0 {dist}\n")
+
+        return photons_file, stars_file, log_file
+
+    def _write_run_script(self, photons_file, stars_file, log_file):
+        """Generate a run script for sim_telarray with the provided configuration and inputs.
+
+        Parameters
+        ----------
+        photons_file, stars_file, log_file : pathlib.Path
+            Input/output files for the run.
+
+        Returns
+        -------
+        pathlib.Path
+            Path to the generated shell script.
+        """
+        script_path = self.scripts_dir / f"run_incident_angles_{self._label_suffix()}.sh"
+        simtel_bin = self._simtel_path / "sim_telarray/bin/sim_telarray_debug_trace"
+        corsika_dummy = self._simtel_path / "sim_telarray/run9991.corsika.gz"
+
+        theta = self.ZENITH_ANGLE_DEG
+        off = float(self.config_data["off_axis_angle"].to_value(u.deg))
+        star_photons = self.config_data["number_of_photons"]
+
+        def cfg(par, val):
+            return f"-C {par}={val}"
+
+        opts = [
+            f"-c {self.telescope_model.config_file_path}",
+            f"-I{self.telescope_model.config_file_directory}",
+        ]
+        if self.perfect_mirror:
+            opts += [
+                "-DPERFECT_DISH=1",
+                "-C random_focal_length=0",
+                "-C mirror_reflection_random_angle=0",
+                "-C mirror_align_random_distance=0",
+                "-C mirror_align_random_horizontal=0,28,0,0",
+                "-C mirror_align_random_vertical=0,28,0,0",
+            ]
+
+        if self.mirror_reflection_random_angle is not None:
+            opts.append(cfg("mirror_reflection_random_angle", self.mirror_reflection_random_angle))
+        if self.mirror_alignment_random is not None:
+            opts.append(
+                cfg("mirror_align_random_horizontal", f"{self.mirror_alignment_random},28.,0.0,0.0")
+            )
+            opts.append(
+                cfg("mirror_align_random_vertical", f"{self.mirror_alignment_random},28.,0.0,0.0")
+            )
+
+        opts += [
+            cfg("IMAGING_LIST", str(photons_file)),
+            cfg("stars", str(stars_file)),
+            cfg("altitude", self.site_model.get_parameter_value("corsika_observation_level")),
+            cfg("telescope_theta", theta + off),
```

Comment:
Do I understand it correctly is e.g. zenith is 45 deg, we simulate [45 deg + offset]? Is this is an issue for telescopes with non-assymmetric response due to shadowing?

Add this point we don't have this in the raytracine, but as Konrad discussed recently we will need that for the SSTs. I want to make sure that we don't simulate here exactly following one of the support beams.